glean-core/ios/Glean/Net/HttpPingUploader.swift

```diff
@@ -6,13 +6,18 @@
 ///
 /// This will typically be invoked by the appropriate scheduling mechanism to upload a ping to the server.
 public class HttpPingUploader {
+    var config: Configuration
+
     // This struct is used for organizational purposes to keep the class constants in a single place
     struct Constants {
+        // Since ping file names are UUIDs, this matches UUIDs for filtering purposes
+        static let filePattern = "[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}"
         static let logTag = "glean/HttpPingUploader"
+        static let pingsDir = "pings"
     }
 
-    public init() {
-        // intentionally left empty to allow public instantiation
+    public init(configuration: Configuration) {
```

Comment:
Do we need the full configuration object?
Probably makes things easier to use to just pass it in fully, but what about only storing the things in this struct that we actually use?

Reply:
I think we are down to just simple types in the Configuration struct now, so along with the timeout, I'll see if there are any other things in the configuration struct that could be removed.